docs/source/index.rst

```diff
@@ -135,6 +135,7 @@ Documentation
    dev/input_processing/model_inputs_index
    dev/multimodal/multimodal_index
    dev/dockerfile/dockerfile
+   dev/profiling/profiling_index
```

Comment:
Seems not necessary to have `_index` as your page is not actually "index". Or you could create 2 pages: profiling_index.rst that introduces all kinds of profiling in vLLM (and let others contribute), and torch_profiler.rst that introduces the method of this PR.

Reply:
+1